src/Workspaces/Core/Portable/EmbeddedLanguages/DateAndTime/LanguageServices/DateAndTimePatternDetector.cs

```diff
@@ -133,32 +137,41 @@ private static bool IsMethodArgument(SyntaxToken token, ISyntaxFacts syntaxFacts
 
         public bool IsDateAndTimeToken(SyntaxToken token, CancellationToken cancellationToken)
         {
```

Comment:
this turned out to be super confusing.  why not just have the core method be: IsDateAndTimeToken(...) => IsDateAndTimeTokenInInvocation(...) || IsDateAndTimeTokenInFormatClause(...)`.  Then pull all the existing logic into the new IsDateAndTimeTokenInInvocation (unchanged otherwise), and keep the other check independent?

Reply:
note that that's what you did for IsPossiblyDateAndTimeToken which was very clear.